diff --git a/main.py b/main.py
index 315483a..3f11b22 100644
--- a/main.py
+++ b/main.py
@@ -11,6 +11,11 @@ import subprocess
 
 
 
+def colorize_number(n):
+    #print(n, type(n))
+    lookup = {0: "0", 2: "231", 4: "230", 8: "216", 16: "214", 32: "208", 64: "196", 128: "228", 256: "227", 512: "226", 1024: "220", 2048: "11", 4096: "232"} 
+    return "\u001b["+lookup[n]+"m\u001b[0m"
+
 class Game:
     def __init__(self):
         """ Initialize a 2048 game
@@ -45,8 +50,9 @@ class Game:
 
     def string(b):
         """ String to pretty print the board in matrix form """
-        return '\n'.join([''.join(['{:8}'.format(item) for item in row])
-                                for row in b])
+        #return '\n'.join([''.join(['{:8}'.format(item) for item in row for row in b])
+
+        return '\n'.join([''.join(['{:8}'.format(colorize_number(item) for item in row for row in b)])])
 
     def spawn(b, k=1):
         """ Add k random tiles to the board.
